### ISSUE TITLE
Is/can format strings benefit from interpolated string optimizations?

### ISSUE THREAD

**Issue by iSazonov** [dotnet/runtime#59567]
After #50601, #50635, #51962 we can

```c#
string message = string.Format(CultureInfo.CurrentCulture, "Current path: {0}", path);
```
convert to:
```c#
string message = string.Format(CultureInfo.CurrentCulture, $"Current path: {path}");
```
and my question is about follow common pattern:
```c#
string message = string.Format(CultureInfo.CurrentCulture, ResourceManager.GetString("Current_Path"), path);
```
where ResourceManager returns a localized version of  "Current path: {0}".

Since ResourceManager returns the same string (until we change current culture that is very rarely) we could parse the format string once and cache/compile. Also in compile time we know all about argument and their types and I guess we could do the same optimizations as for interpolated strings.

In PowerShell we have a lot of such patterns with format strings from resources and only a few ones as static (which could be converted to interpolated string already today) and the optimization I ask could be very useful.

(I found similar https://github.com/dotnet/csharplang/issues/2347 but that discussion was in another context.)

**Comment by stephentoub** [comment]
Duplicate of https://github.com/dotnet/runtime/issues/50330

**Comment by iSazonov** [comment]
@stephentoub Thanks for pointing #50330. I think #50330 (as well as https://github.com/dotnet/csharplang/issues/2347) discussion goes a little in a different direction. My reasoning comes from PowerShell.

PowerShell uses _strong typed resources_. This means that there is already a code generated at (before :-) ) compile time. But the code is still a wrapper around traditional resource strings.
Traditional resource strings are de-facto constants _known at compile time_. So we need to do next step and  convert the resource strings to a code. 
(Not all resource strings are an interpolated strings. So perhaps we need to explicitly annotate them as interpolated strings or/and put in new source resource file.)

Thus
```c#
string.Format(culture, FileSystemStrings.PathNotFound, path);
```
will be compiled to:
```c#
FileSystemStrings.PathNotFound(culture, path);
```
with (simplified) implementation:
```c#
class FileSystemStrings
{
    public PathNotFound(CultureInfo culture, string path)
    {
        string.Format(culture, $"Path not found: {path}.");
    }
}
```

Of course we need an analog of ResourceManager class to discover culture specific assembly, load it and cache result by culture as ResourceManager does.

I guess C# compiler could be so smart and does the work for us if FileSystemStrings.PathNotFound parameter would be annotated with new string interpolated attribute. At the same time to convert the resource file is probably better with a source generator.

This, of course, is not the final sentence, but just the initial considerations.
